### PR TITLE
docs: update French README

### PR DESCRIPTION
**Pull request by daank-c**
Updates the French README to reflect the new branching strategy introduced in version 1.14.8. This update follows the changes made at commit `9222634` and merge `70e7bf6`.

### CODE REVIEW

**Comment by patricklodder** [comment]
Added `help wanted` for review by a native speaker

**Comment by John-Inubook** [comment]
I can review the French version.

**Review by John-Inubook — APPROVED** [review]
All good 👍

File: doc/intl/fr/README.md [hunk 1]
```diff
@@ -43,20 +45,25 @@ Principales ressources de développement :
...
+
 Les numéros de version suivent la sémantique ```major.minor.patch```.
 
 ### Branches
-Il y a 3 types de branches dans ce référentiel :
 
-- **master:** Stable, contient la dernière version de la dernière *major.minor* version.
-- **maintenance:** Stable, contient la dernière version des versions précédentes, qui sont toujours en maintenance active. Format : ```<version>-maint```
-- **development:** Instable, contient du nouveau code pour les versions prévues. Format : ```<version>-dev```
+Il y a 4 types de branches dans ce référentiel :
```
Comment (loom4k) [inline doc/intl/fr/README.md:57]: "Dans ce référentiel" is the literal translation of "in this repository" but it's not grammatically correct. I would suggest phrasing this as follow:
```suggestion
Dans ce projet se trouvent 4 types de branches :
```
or even
```suggestion
Il y a 4 types de branches dans ce projet
```

Référentiel is a french word for reference. It has no direct link to a repository and a native french speaker would not think of "référentiel" as being a repository. Reading this would actually correctly translate to: "There are 4 types of branches under these references :"

File: doc/intl/fr/README.md [hunk 2]
```diff
@@ -43,20 +45,25 @@ Principales ressources de développement :
...
+
+- **master :** Instable, contient le code le plus récent en développement.
+- **maintenance :** Stable, contient la dernière version des versions précédentes, qui sont toujours en maintenance active. Format : ```<version>-maint```
+- **development :** Instable, contient du nouveau code pour les versions prévues. Format : ```<version>-dev``` 
+- **archive :** Stable, branches immuables pour les anciennes versions qui ne changent plus car elles ne sont plus maintenues.
```
Comment (loom4k) [inline doc/intl/fr/README.md:62]: Saying "immuable" and "qui ne changent plus" is a pleonasm. I would rephrase this as follow:

```suggestion
- **archive :** Stable, branches immuables d'anciennes versions qui ne sont plus maintenues.
```

File: doc/intl/fr/README.md [hunk 3]
```diff
@@ -11,6 +11,8 @@ Dogecoin Core [DOGE, Ð]
 
 </div>
 
+IMPORTANT : À partir d'août 2024, la branche master est devenue la branche principale d'intégration et est devenue instable. Veuillez consulter une version taguée avant de compiler des binaires de production
```
Comment (loom4k) [inline doc/intl/fr/README.md:14]: I would put "taguée" in italic since it is not a real french word.

**Review by loom4k — CHANGES_REQUESTED** [review]
I'm a native French speaker and thought I would suggest some changes. This is overall really good but some words don't exist or the phrasing is incorrect. Below are my suggestions.

**Comment by loom4k** [comment]
> All good 👍

@John-Inubook this is in fact (in my opinion) not all good. Some phrases are grammatically incorrect and some words don't even refer to the correct English translation. I suggested some changes to be made.

File: doc/intl/fr/README.md (hunk 1 above)
Comment (loom4k) [reply doc/intl/fr/README.md:57]: That is very strange. I doubt any native French speaker would find a similarity between Référentiel and Repository. Is it possible GitHub used AI or Google-like-translate to translate some of its features? Or maybe someone on a Crowdin-like-platform used a Google-like-translate to put in that word?

When I entered Repository on Google translate I got back Référentiel so it would make sense in my mind if that's how the word got there in the first place.

I find project to be a more visual word for repository than référentiel and also simpler to understand. Let's leave it up to the reviewers to decide which one they like more I suppose.

Comment (loom4k) [reply doc/intl/fr/README.md:57]: @daank-c Look at how they translated "Pull requests" as "Demande de tirage". This makes absolutely no sense and is definitely not the correct translation. I suppose an online translator or the use of AI. The docs should definitely not use Référentiel.

**Comment by patricklodder** [comment]
Thanks for contributing everyone.

> John-Inubook this is, in my opinion, not all good

Reviews complement each other; the more eyes on work, the more opportunity there is to find improvements - so no worries we'll leave space to get to some consensus. Either on this pull request, or through subsequent pull requests - whatever it takes.

File: doc/intl/fr/README.md (hunk 1 above)
Comment (John-Inubook) [reply doc/intl/fr/README.md:57]: Repository would be translated to répertoire which would work. "Référentiel" is by definition a system of reference, which is what a repository is. So if the word "référentiel" doesn't sound right, I guess Répertoire would be good too.

Comment (loom4k) [reply doc/intl/fr/README.md:57]: I think Répertoire would actually be more suited to refer to a directory.

File: doc/intl/fr/README.md (hunk 3 above)
Comment (John-Inubook) [reply doc/intl/fr/README.md:14]: Taguer is indeed a French verb : Définition de taguer ​​​ verbe transitif
au participe passé Murs tagués. Mentionner le nom de (qqn, qqch.) dans un tag (2). Taguer ses amis sur une photo.

File: doc/intl/fr/README.md (hunk 2 above)
Comment (John-Inubook) [reply doc/intl/fr/README.md:62]: Immuables is referring to branches and qui ne changent plus is referring to les anciennes versions, so each are referring to a specific and distinctive word. But the change you're suggesting does sound better.

File: doc/intl/fr/README.md (hunk 3 above)
Comment (loom4k) [reply doc/intl/fr/README.md:14]: We can keep it. I would still put it in italic as it's not exactly the same definition.

**Review by loom4k — APPROVED** [review]
Thanks for the modifications. This looks all good now. Could be merged whenever. Awesome work @daank-c!

**Review by John-Inubook — APPROVED** [review]
Perfect 👌

**Comment by loom4k** [comment]
@daank-c I suggest we use the same folder structure as the other localization packages and rename the `intl/fr/...` directory to `intl/fr_FR/...`. Others (including me in https://github.com/dogecoin/dogecoin/pull/3624) might want to share some aspects of the `FR` folder. What do you think?

**Comment by loom4k** [comment]
> @loom4k idk, I think each directory should be consistent with the naming convention used in `src/qt/locale/`. Also, where possible, we should try to limit ourselves to one dialect per language, so we only have to maintain one file.
> 
> Might not always work, but worth trying to achieve as it minimizes the maintenance burden.

@daank-c We are saying the same thing. Renaming it to `fr_FR` would keep it consistent as the capitalized `FR` is the country (France) and the lowercase one is the language (French). So, we're modifying it to French of France. At some point there might be a French from Canada, and what I'm trying to add is Breton from France (which is a totally different language not a dialect). If you look at the other folders that's what they are doing (especially Portuguese either from Brazil or Portugal).

If you prefer, we could keep it as `fr` like you have it (if we think no integration of Canadian French might be made) and I'll rename my `bzh_FR` to simply `bzh`. The only issue I might have with this is that `bzh` doesn't really indicate what country it's from so some confusion could occur. Your call 😉

**Comment by patricklodder** [comment]
> If you prefer, we could keep it as fr like you have it 

Let's keep it like `fr`. Dialects are fun but add to maintenance pressure. We've discussed this many times before, also for `fr` - if there is a generic version of French possible that is understandable to the majority of French speakers (unlike with Portuguese) then this version should be that.

Reach as many people as possible with as little work as possible => most efficient way and we don't waste people's time.

**Review by patricklodder — APPROVED** [review]
utACK - translation verified by @John-Inubook and @loom4k
